### PR TITLE
added components for the product,added  search and filter feature

### PR DESCRIPTION
**Pull request by folusho-adeyemi**
## Description

Adding the search functionality
Added the filtering feature

## Milestones
Milestone 2, [1a-1b]

## Resources
Inspirations from my previous projects

## Test Plan
Screenshot 2023-07-10 at 4 30 19 PM
Screenshot 2023-07-10 at 4 30 24 PM
Screenshot 2023-07-10 at 4 30 34 PM

<any edge cases you might have specifically tested>

### CODE REVIEW

File: FashionConnect_ui/src/components/Main/Main.jsx [hunk 1]
```diff
@@ -1,12 +1,25 @@
...
 import NavBar from "../NavBar/NavBar";
+import { useState ,useEffect } from "react";
 import Products from "../Products/Products";
 
 export default function Main() {
+    const [products, setProducts] = useState([]);
+
+    useEffect(() => {
+        const fetchProducts = async () => {
+        const response = await fetch('http://localhost:3000/products');
```
Comment (safirshahbaaz) [inline FashionConnect_ui/src/components/Main/Main.jsx:11]: You can create a constant of this URL in your file for reference in other places if needed.

File: FashionConnect_ui/src/components/ProductCard/ProductCard.jsx [hunk 2]
```diff
@@ -0,0 +1,17 @@
+import "./ProductCard.css";
+import React from "react";
+
+export default function ProductCard({product}){
```
Comment (safirshahbaaz) [inline FashionConnect_ui/src/components/ProductCard/ProductCard.jsx:4]: Nice, I like having different components for the different parts, good job!

File: FashionConnect_ui/src/components/Products/Products.jsx [hunk 3]
```diff
@@ -1,29 +1,96 @@
...
+            </form>
+        </div>
+        <div className="menu">
+            <li>
+            <button className="menu-btn" onClick={() => handleClick("")}>
+                All Categories
+            </button>
+            </li>
+            <li>
+            <button className="menu-btn" onClick={() => handleClick("Tops")}>
```
Comment (safirshahbaaz) [inline FashionConnect_ui/src/components/Products/Products.jsx:42]: Once you have the data available in your database, you should replace this strings with the categoryID. That way, you can turn this entire logic into function calls that returns the button component with the right text.

Comment (safirshahbaaz) [reply FashionConnect_ui/src/components/Products/Products.jsx:42]: You can also consider doing this now too with the strings and see how your code will shrink to a few lines. :)

**Review by safirshahbaaz — APPROVED** [review]
Overall looks good!

File: FashionConnect_ui/src/components/Main/Main.jsx (hunk 1 above)
Comment (folusho-adeyemi) [reply FashionConnect_ui/src/components/Main/Main.jsx:11]: Ok

File: FashionConnect_ui/src/components/ProductCard/ProductCard.jsx (hunk 2 above)
Comment (folusho-adeyemi) [reply FashionConnect_ui/src/components/ProductCard/ProductCard.jsx:4]: Yeah, I found out that it makes my code more readable.

File: FashionConnect_ui/src/components/Products/Products.jsx (hunk 3 above)
Comment (folusho-adeyemi) [reply FashionConnect_ui/src/components/Products/Products.jsx:42]: Ok cool!
I already have the data for caretgory Id. I will apply the changes

**Comment by folusho-adeyemi** [comment]
Thank you!
